src/test/java/com/github/susom/database/test/CommonTest.java

```diff
@@ -98,6 +89,20 @@ public void closeJdbc() throws Exception {
     }
   }
 
+  @Test
+  public void tableExists() {
+    String testTableName = (db.flavor().equals(Flavor.postgresql) || db.flavor().equals(Flavor.sqlserver))
+      ? TEST_TABLE_NAME
+      : TEST_TABLE_NAME.toUpperCase(); // Oracle, Derby, & HSQL names are upper case.
```

Comment:
I don't think we want to make this the user's responsibility.  Can we put it into the tableExists method?  Or just have tableExists do a case-insensitive comparison?

Reply:
The  metadata.getTables API expects us to pass in the table name (and optionally schema and catalog) that the user requested to narrow the scope.  That lookup is case sensitive depending on the DB driver. 

To make this work as you request, I would need to modify the call to do a wildcard query on all 3 (something like this: metadata.getTables("%", "%", "%", new String[] {"TABLE", "VIEW"});)and then step through doing a case insenstive match on catalog, schema, and table as a tuple instead of letting the JDBC API handle it as it was designed to do.  Not very pretty.

I can do that if this is what we want, but I'm ambivalent about this kind of overhead if the user has a large database with a lot of catalogs, schemas, and tables.  Garrick, can you weigh in on this?

Reply:
my 2 cents would be to not hide the idiosyncrasies of each rdbms from the user as he/she will have to deal with it in the SQL world anyways. it would be very confusing to see it work in susom and not in SSMS or Sql Developer.

Reply:
@smalunjk I'd say one of the purposes of the database library is to hide the idiosyncrasies of the individual implementations.  That's why the Flavor type exists.
@glenna-m I agree that there's no good way to do a case-insensitive query with getTables, which says the table name pattern "must match the table name as it is stored in the database".  But every database normalizes unquoted names to either upper case or lower case, and if we make that part of the Flavor information, we can use it here.  It would need a warning that tableExists may not work for quoted table names that override the default case normalization.  Since it's a new API, that doesn't seem like a serious restriction, and also anybody using case-sensitive table names should reconsider the life-choices that led them to this point 😁 .